libmscore/layout.cpp

```diff
@@ -1976,162 +1986,160 @@ void Score::createMMRest(Measure* m, Measure* lm, const Fraction& len)
             }
         }
         if (!found) {
-            mmr->add(e->clone());
+            mmrMeasure->add(e->clone());
         }
     }
     for (Element* e : oldList) {
         delete e;
     }
-    Segment* s = mmr->undoGetSegmentR(SegmentType::ChordRest, Fraction(0,1));
+    Segment* s = mmrMeasure->undoGetSegmentR(SegmentType::ChordRest, Fraction(0,1));
     for (int staffIdx = 0; staffIdx < _staves.size(); ++staffIdx) {
         int track = staffIdx * VOICES;
         if (s->element(track) == 0) {
-            Rest* r = new Rest(this);
-            r->setDurationType(TDuration::DurationType::V_MEASURE);
-            r->setTicks(mmr->ticks());
-            r->setTrack(track);
-            r->setParent(s);
-            undo(new AddElement(r));
+            MMRest* mmr = new MMRest(this);
+            mmr->setDurationType(TDuration::DurationType::V_MEASURE);
+            mmr->setTicks(mmrMeasure->ticks());
+            mmr->setTrack(track);
+            mmr->setParent(s);
+            undo(new AddElement(mmr));
         }
     }
 
     //
-    // check for clefs
+    // further check for clefs
     //
-    Segment* cs = lm->findSegmentR(SegmentType::Clef, lm->ticks());
-    Segment* ns = mmr->findSegment(SegmentType::Clef, lm->endTick());
-    if (cs) {
-        if (ns == 0) {
-            ns = mmr->undoGetSegmentR(SegmentType::Clef, lm->ticks());
-        }
-        ns->setEnabled(cs->enabled());
-        ns->setTrailer(cs->trailer());
+    Segment* underlyingSeg = lastMeasure->findSegmentR(SegmentType::Clef, lastMeasure->ticks());
+    Segment* mmrSeg = mmrMeasure->findSegment(SegmentType::Clef, lastMeasure->endTick());
+    if (underlyingSeg) {
+        if (mmrSeg == 0) {
+            mmrSeg = mmrMeasure->undoGetSegmentR(SegmentType::Clef, lastMeasure->ticks());
+        }
+        mmrSeg->setEnabled(underlyingSeg->enabled());
+        mmrSeg->setTrailer(underlyingSeg->trailer());
         for (int staffIdx = 0; staffIdx < _staves.size(); ++staffIdx) {
             int track = staffIdx * VOICES;
-            Clef* clef = toClef(cs->element(track));
+            Clef* clef = toClef(underlyingSeg->element(track));
             if (clef) {
-                if (ns->element(track) == 0) {
-                    ns->add(clef->clone());
+                if (mmrSeg->element(track) == 0) {
+                    mmrSeg->add(clef->clone());
                 } else {
                     //TODO: check if same clef
                 }
             }
         }
-    } else if (ns) {
+    } else if (mmrSeg) {
         // TODO: remove elements from ns?
```

Comment:
One day something will go wrong where we finally need to understand what is going on here...